go.mod

```diff
@@ -22,32 +22,32 @@ replace (
-	k8s.io/apiserver => k8s.io/apiserver v0.24.5
-	k8s.io/cli-runtime => k8s.io/cli-runtime v0.24.5
-	k8s.io/client-go => github.com/rancher/client-go v1.24.0-rancher1
-	k8s.io/cloud-provider => k8s.io/cloud-provider v0.24.5
-	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.24.5
-	k8s.io/code-generator => k8s.io/code-generator v0.24.5
-	k8s.io/component-base => k8s.io/component-base v0.24.5
-	k8s.io/component-helpers => k8s.io/component-helpers v0.24.5
-	k8s.io/controller-manager => k8s.io/controller-manager v0.24.5
-	k8s.io/cri-api => k8s.io/cri-api v0.24.5
-	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.24.5
-	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.24.5
-	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.24.5
-	k8s.io/kube-proxy => k8s.io/kube-proxy v0.24.5
-	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.24.5
-	k8s.io/kubectl => k8s.io/kubectl v0.24.5
-	k8s.io/kubelet => k8s.io/kubelet v0.24.5
-	k8s.io/kubernetes => k8s.io/kubernetes v1.24.2
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.24.5
-	k8s.io/metrics => k8s.io/metrics v0.24.5
-	k8s.io/mount-utils => k8s.io/mount-utils v0.24.5
-	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.24.5
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.24.5
+	k8s.io/api => k8s.io/api v0.25.4
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.25.4
+	k8s.io/apimachinery => k8s.io/apimachinery v0.25.4
+	k8s.io/apiserver => k8s.io/apiserver v0.25.4
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.25.4
+	k8s.io/client-go => github.com/rancher/client-go v1.25.4-rancher1
+	k8s.io/cloud-provider => k8s.io/cloud-provider v0.25.4
+	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.25.4
+	k8s.io/code-generator => k8s.io/code-generator v0.25.4
+	k8s.io/component-base => k8s.io/component-base v0.25.4
+	k8s.io/component-helpers => k8s.io/component-helpers v0.25.4
+	k8s.io/controller-manager => k8s.io/controller-manager v0.25.4
+	k8s.io/cri-api => k8s.io/cri-api v0.25.4
+	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.25.4
+	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.25.4
+	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.25.4
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.25.4
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.25.4
+	k8s.io/kubectl => k8s.io/kubectl v0.25.4
+	k8s.io/kubelet => k8s.io/kubelet v0.25.4
+	k8s.io/kubernetes => k8s.io/kubernetes v1.25.4
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.25.4
+	k8s.io/metrics => k8s.io/metrics v0.25.4
+	k8s.io/mount-utils => k8s.io/mount-utils v0.25.4
+	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.25.4
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.25.4
 
 	sigs.k8s.io/aws-iam-authenticator => github.com/rancher/aws-iam-authenticator v0.5.9-0.20220713170329-78acb8c83863
 	sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.2.0
```

Comment:
Do we need cluster-api v1.2.2?  It has support for 1.25 https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.2.2. 
I usually bump cluster-api as part of dependency bump for minor version

Reply:
good catch! it is fixed now.

Reply:
@jiaqiluo updating capi also requires updating capi-crds and capi-webhooks(they are in upstream repo) example PR: https://github.com/rancher/rancher/commit/52acb2f60abe7310c4cda75fbeda243bb9181025

Reply:
There are no changes on the c between 1.2.7 and 1.2.0 in the upstream this time, so we are good 😄 
see https://github.com/kubernetes-sigs/cluster-api/compare/v1.2.0...v1.2.7